via/views/view_video.py

```diff
@@ -1,15 +1,18 @@
 from h_vialib import Configuration
 from pyramid.view import view_config
 
+from via.services import YoutubeService
+
 
 @view_config(
     renderer="via:templates/video_player.html.jinja2",
     route_name="view_video",
 )
-def view_video(request):
+def view_video(context, request):
```

Comment:
We take full youtube URLs now instead of the video id and parse it here.

This is the same approach as for PDFs but there we split this further to support different PDF providers. As we only support youtube we don't need that specialization here.

If we were to add other video providers we'd keep the /video route taking the full URL as the main entry point.

Reply:
Ok, there was an earlier PR (https://github.com/hypothesis/via/pull/905) that added this `/video/*` route, but I missed that.

So this is a new video player app that's part of Via now. But my understanding is that we're not actually proxying the bytes of YouTube videos through Via. Rather, Via just renders a page that includes the YouTube video player embedded, the video transcript, and the annotation client.